tech/firewall-rule-format.md

```diff
@@ -575,15 +600,487 @@ For the record, the reason why the protocol is `*` in Azure is that
+
+    {
+      "kind": "compute#firewall",
+      "id": "4387657442666205482",
+      "creationTimestamp": "2019-03-26T11:30:29.863-07:00",
+      "name": "allow-8000",
+      "description": "",
+      "network": "https://www.googleapis.com/compute/v1/projects/strong-augury-224506/global/networks/default",
+      "priority": 1000,
+      "sourceRanges": [
+        "0.0.0.0/0"
+      ],
+      "allowed": [
+        {
+          "IPProtocol": "tcp",
+          "ports": [
+            "8000"
+          ]
+        }
+      ],
+      "direction": "INGRESS",
+      "logConfig": {
+        "enable": false
+      },
+      "disabled": false,
+      "selfLink": "https://www.googleapis.com/compute/v1/projects/strong-augury-224506/global/firewalls/allow-8000"
+    }
+
+
+### Any Source: Conclusion
+
+Once we have identified that a critical port such as `22` or `3389` is
+exposed as discussed in the previous section, we need to identify if it
+is exposed to the entire Internet.
+
+In this section, we learn that checking this is not straightforward in
+case of Azure. There are multiple formats to consider in Azure:
+
+  - `"source_address_prefix": "*"`
+  - `"source_address_prefix": "0.0.0.0/0"`
+  - `"source_address_prefix": "Internet"`
+  - `"source_address_prefixes"` list contains `0.0.0.0/0`
+
+With GCP, it is straightforward due to its simple UI and simple JSON
+structure. We only need to check if `0.0.0.0/0` exists in
+`sourceRanges`.
+
+One way to simplify this in Azure could be to normalize the source
+ranges in `source_address_prefix` and `source_address_prefixes` to a
+common format, e.g., normalize both `*` and `Internet` to `0.0.0.0/0`
+and let the source ranges in the common (`com`) bucket be always a list
+even if a single address in `source_address_prefix` is available.
+
+
+Any Port
+--------
+
+### Any Port: Azure
+
+In this section, we will see a couple of ways we can allow connections
+to all ports.
+
+
+#### `destination_port_range` set to `*`
+
+Here is an example where an inbound security rule is created with
+*Destination port ranges* set to `*`.
+
+    {
+      "id": "/subscriptions/a8daa361-b951-46b5-8e9d-b375f2ffe9fd/resourceGroups/rg1/providers/Microsoft.Network/networkSecurityGroups/vm1-nsg/securityRules/Port_All",
+      "protocol": "*",
+      "source_port_range": "*",
+      "destination_port_range": "*",
+      "source_address_prefix": "*",
+      "source_address_prefixes": [],
+      "destination_address_prefix": "*",
+      "destination_address_prefixes": [],
+      "source_port_ranges": [],
+      "destination_port_ranges": [],
+      "access": "Allow",
+      "priority": 100,
+      "direction": "Inbound",
+      "provisioning_state": "Succeeded",
+      "name": "Port_All",
+      "etag": "W/\"2873cc0e-f41e-4c94-a773-8b0982d0db55\""
+    }
+
+
+#### `destination_port_range` set to integer range
+
+Another way to allow access to all ports is to set *Destination port
+ranges* to `0-65535`.
```

Comment:
The key here is also `destination_port_range`.

Reply:
See https://github.com/cloudmarker/guides/pull/5#discussion_r270779322.